pyacq/core/tests/test_tools.py

```diff
@@ -0,0 +1,70 @@
+from pyacq.core.stream  import OutputStream, InputStream
+from pyacq.core.tools  import ThreadPollInput
+from pyqtgraph.Qt import QtCore, QtGui
+import pyqtgraph as pg
+
+import numpy as np
+import weakref
+import time
+
+
+def test_ThreadPollInput():
+    app = pg.mkQApp()
+    
+    nb_channel = 16
+    chunksize = 1024
+    stream_spec = dict(protocol = 'tcp', interface = '127.0.0.1', port='*', 
+                       transfermode = 'plaindata', streamtype = 'analogsignal',
+                       dtype = 'float32', shape = (-1, nb_channel), compression ='',
+                       scale = None, offset = None, units = '', timeaxis = 0)
+    
+    class ThreadSender(QtCore.QThread):
+        def __init__(self, output_stream, parent = None):
+            QtCore.QThread.__init__(self, parent)
+            self.output_stream = weakref.ref(output_stream)
```

Comment:
I think the zmq socket should belong to the thread that is reading it. Ideally the socket would be created in the same thread, but since that is sometimes difficult, a compromise would be to create the socket in one thread, pass it to the poller thread, and then delete the reference from the original thread, to avoid any possibility of touching the socket from the wrong thread.

For example: zmq sockets probably do some cleanup when they are collected. If this cleanup happens in one thread while the socket is being polled in another, then you might crash.